s3/s3_operations.cpp

```diff
@@ -245,19 +245,44 @@ namespace irods_s3 {
             }
         }
 
-
-        // if this is a replication and we're the destination, get the data size from the source dataObjInfo
+        // special treatment for replication - get data_size from source and
+        // number_of_threads from destination
```

Comment:
Consider tweaking this comment so that it explains **why** we do this.

Reply:
done